node/impl/paych/paych.go

```diff
@@ -40,6 +40,10 @@ func (a *PaychAPI) PaychGet(ctx context.Context, from, to address.Address, ensur
 	}, nil
 }
 
+func (a *PaychAPI) PaychGetWaitReady(ctx context.Context, mcid cid.Cid) (address.Address, error) {
```

Comment:
Possible suggestion: this should take a ChannelID as well, or the whole ChannelInfo struct.

The reason I'd suggest for that is that if I got a message CID from someone else -- but not one I created through PaychGet, I might call this thinking it will work, but absent that message being in the database it won't. If I require a channelID / ChannelInfo I know I need then to reference a message cid I  got from paych Get

Reply:
alternatively, just use a UUID instead?

Reply:
by this I mean maybe api.Channelnfo has a ChannelMessage that is a UUID, not an actual message CID, and oyu put the message CID int the datastore -- that would prevent people using external message CIDs with this call and also prevent using StateWaitMessage with the channelInfo.ChannelMessage

Reply:
These are good points 👍 

I think the issue is really a UX issue: if we return a message CID, the client might figure they can use a message CID from someone else when calling PaychGetWaitReady().

My suggestion would be to return a value called "waitID" that is just a stringified version of the message CID. That way the client doesn't think it's a message CID, but we can still store a single value (instead of having to store a separate wait ID).

Reply:
sounds good.

Reply:
https://github.com/filecoin-project/lotus/pull/2986